libfreerdp/gdi/gdi.c

```diff
@@ -594,9 +600,15 @@ static void gdi_patblt(rdpContext* context, PATBLT_ORDER* patblt)
 		originalBrush = gdi->drawing->hdc->brush;
 
 		gdi->drawing->hdc->brush = gdi_CreateSolidBrush(foreColor);
+		if (!gdi->drawing->hdc->brush)
+		{
+			ret = FALSE;
+			goto out_error;
+		}
 
-		gdi_PatBlt(gdi->drawing->hdc, patblt->nLeftRect, patblt->nTopRect,
-				patblt->nWidth, patblt->nHeight, gdi_rop3_code(patblt->bRop));
+		if (gdi_PatBlt(gdi->drawing->hdc, patblt->nLeftRect, patblt->nTopRect,
+				patblt->nWidth, patblt->nHeight, gdi_rop3_code(patblt->bRop)) != 1)
```

Comment:
This is ok for freerdp, however I'd suggest to check for == 0 to match the msdn description.